Bottom: 28ab3bbc9185c120977b1a13fe304663346a51d2
Top:    bc342f2b0690e08f967193c92488aeaebc1ff6ce
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-26 14:23:41 +0100

draw.c: Draw pins, pads and vias (for non-gui HIDs) from DrawLayerGroup()


---

diff --git a/src/draw.c b/src/draw.c
index 46b3063..697f676 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -366,7 +366,6 @@ PrintAssembly (int side, const BoxType * drawn_area)
 
   gui->set_draw_faded (Output.fgGC, 1);
   DrawLayerGroup (side_group, drawn_area);
-  DrawPPV (side_group, drawn_area);
   gui->set_draw_faded (Output.fgGC, 0);
 
   /* draw package */
@@ -430,8 +429,7 @@ DrawEverything (BoxTypePtr drawn_area)
 
       if (gui->set_layer (0, group, 0))
         {
-          if (DrawLayerGroup (group, drawn_area) && !gui->gui)
-            DrawPPV (group, drawn_area);
+          DrawLayerGroup (group, drawn_area); 
           gui->end_layer ();
         }
     }
@@ -788,7 +786,9 @@ DrawLayerGroup (int group, const BoxType *drawn_area)
     }
   if (n_entries > 1)
     rv = 1;
-  return rv;
+
+  if (rv == 1 && !gui->gui)
+    DrawPPV (group, drawn_area);
 }
 
 static void
